ui/src/app/pages/data/concept/concept-table.tsx

```diff
@@ -134,8 +142,22 @@ export class ConceptTable extends React.Component<Props, State> {
     }
   }
 
-  updateSelectedConceptList(selectedConcepts) {
-    this.setState({selectedConcepts : selectedConcepts});
+  updateSelectedConceptList(selectedConcepts, origin) {
+    // By default Data table will select all the concepts in the table but since we have first give
+    // an option to user to select all concepts in a page
+    // we will just add the the concepts in the page to selected concept list
+    if (selectedConcepts.length === this.props.concepts.length && origin === 'table') {
+      const startIndex = this.state.tableRef.current.state.first;
+      const endIndex = startIndex + ROWS_TO_DISPLAY;
+      const allSelectedConcepts = this.state.selectedConcepts
+          .concat(this.props.concepts.slice(startIndex, endIndex));
+      selectedConcepts = fp.uniqBy( 'conceptId', allSelectedConcepts);
```

Comment:
Opt nit: I might cut out out the intermediate variable and just put the concat statement here.